addOns/ascanrulesBeta/src/main/java/org/zaproxy/zap/extension/ascanrulesBeta/InsecureHttpMethodScanRule.java

```diff
@@ -642,4 +622,80 @@ private void testHttpMethod(String httpMethod) throws Exception {
     private static String randomAlphanumeric(int count) {
         return RandomStringUtils.secure().nextAlphanumeric(count);
     }
+
+    private AlertBuilder buildAlert(
+            String vulnName,
+            String vulnDesc,
+            String extraInfo,
+            String evidence,
+            HttpMessage msg,
+            VulnType currentVT) {
+        return newAlert()
+                .setConfidence(Alert.CONFIDENCE_MEDIUM)
+                .setName(
+                        Constant.messages.getString(
+                                "ascanbeta.insecurehttpmethod.detailed.name", vulnName))
+                .setDescription(vulnDesc)
+                .setOtherInfo(extraInfo)
+                .setSolution(Constant.messages.getString("ascanbeta.insecurehttpmethod.soln"))
+                .setEvidence(evidence)
+                .setMessage(msg)
+                .setCweId(getCweId())
+                .setWascId(getWascId())
+                .setAlertRef(getId() + "-" + currentVT.getRef());
+    }
```

Comment:
The builder should be usable by the existing code. \Ex: Where alerts are currently raised it should be able to call the builder method(s) (there can be multiple if necessary) then just raise, like: `buildAlert(....).raise()`;`